src/Microsoft.Health.Api/Features/ByteCounter/HeaderUtility.cs

```diff
@@ -0,0 +1,41 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Text;
+using EnsureThat;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Health.Api.Features.ByteCounter
+{
+    public static class HeaderUtility
+    {
+        private static readonly Encoding HeaderEncoding = Encoding.UTF8;
```

Comment:
Is it guaranteed headers are always UTF-8?

Reply:
I did a bit of research and it seems this should work for most cases, but I update the HeaderUtility so it specifically asks for an encoding. That way it will be explicit if someone else uses it with a special case